### PR TITLE
Resize for slow loading images

### PR DESCRIPTION
**Pull request by divyaashritha**
Issue : https://github.com/cerner/xfc/issues/35

### CODE REVIEW

File: src/provider/application.js [hunk 1]
```diff
@@ -23,14 +23,17 @@ class Application extends EventEmitter {
     this.secret = secret;
     this.onReady = onReady;
     this.targetSelectors = targetSelectors;
-    this.resizeConfig = null;
+    this.resizeConfig = {};
```
Comment (mhemesath) [inline src/provider/application.js]: Actually thinking about this. We should probably leave this null. Then in "imageRequestResize" if resize config isn't set it should do nothing. This means the SDK hasn't done initial resize and will spare computation to redundantly do it for the image and then again for SDK as a whole.

Comment (mhemesath) [reply src/provider/application.js]: alternative is to make `requestResize` return out iif config isn't set

File: test/application.js [hunk 2]
```diff
@@ -49,6 +49,27 @@ describe('Application', () => {
     it("sets application's JSONRPC", () => {
       expect(application.JSONRPC).to.be.an.instanceof(JSONRPC);
     });
+    
+    it("calls addEventListener", sinon.test(function() {
+      sinon.spy(document.body, 'addEventListener');
+      expect(document.body.addEventListener.calledOnce);
+    }));
+
+    describe('#imageRequestResize(event)', () => {
```
Comment (alex-bezek) [inline test/application.js]: this doesn't seem like it should be in the init describe, but its a bit hard to trrack all the blocks start/end on this PR

File: test/application.js [hunk 3]
```diff
@@ -49,6 +49,27 @@ describe('Application', () => {
...
+        application.resizeConfig = {}
+        const event = {
+          target: { 
+            tagName: "IMG", 
+            hasAttribute: (attr) => { attr === 'height'}
+          }
+        };  
+        application.imageRequestResize(event);
+
+        expect(application.requestResize.calledOnce);
```
Comment (alex-bezek) [inline test/application.js]: we should have a test for each case to show when we expect it to not be called

**Comment by alex-bezek** [comment]
https://github.com/cerner/xfc/blob/master/CHANGELOG.md
should update this too i think

File: src/provider/application.js [hunk 4]
```diff
@@ -76,6 +79,17 @@ class Application extends EventEmitter {
     );
   }
 
+  /**
+   * imageRequestResize function to call requestResize event for slow loading image
+   * @param {object} event - event which triggered the listener
+   */
+  imageRequestResize(event) {
+    const tgt = event.target;
+    if (this.resizeConfig && tgt.tagName === 'IMG' && !(tgt.hasAttribute('height') || tgt.hasAttribute('width'))) {
```
Comment (mhemesath) [inline src/provider/application.js]: resizeConfigi check should go in the requestResize function. It shouldn't execute until its "been initialized" by receiving config

**Pull request merged**